lib/Models/SearchProviders/CatalogIndex.ts

```diff
@@ -0,0 +1,131 @@
+import { ObservableMap, action, observable } from "mobx";
+import CatalogIndexReference from "../Catalog/CatalogReferences/CatalogIndexReference";
+import loadJson from "../../Core/loadJson";
+import updateModelFromJson from "../Definition/updateModelFromJson";
+import CommonStrata from "../Definition/CommonStrata";
+import Terria from "../Terria";
+import CatalogIndexReferenceTraits from "../../Traits/TraitsClasses/CatalogIndexReferenceTraits";
+import SearchResult from "./SearchResult";
+
+const { Document } = require("flexsearch");
+
+export interface CatalogIndexFile {
+  [id: string]: CatalogIndexReferenceTraits;
+}
+
+export interface ModelIndex {
+  name: string;
+  knownContainerUniqueIds: string[];
+}
+
+export default class CatalogIndex {
+  private _models: ObservableMap<string, CatalogIndexReference> | undefined;
```

Comment:
Does this have to be an `ObservableMap`?

Reply:
Oh yeah good pickup, no it doesn't - i've turned it into a normal map